### PR TITLE
fix: deprecated `store::UnorderedMap` and `store::UnorderedSet` due to not meeting the original requirements (iteration over a collection of more than 2k elements runs out of gas)

### PR DESCRIPTION
**Pull request by ruseinov**
- deprecated `store::UnorderedMap` due to its fragmentation property
- deprecated `store::UnorderedSet` due to its fragmentation property

### CODE REVIEW

File: near-sdk/src/store/frangible_map/mod.rs [hunk 1]
```diff
@@ -18,23 +18,28 @@ use super::{FreeList, LookupMap, ERR_INCONSISTENT_STATE, ERR_NOT_EXIST};
...
-/// The default hash function for [`UnorderedMap`] is [`Sha256`] which uses a syscall
+/// The default hash function for [`FrangibleUnorderedMap`] is [`Sha256`] which uses a syscall
 /// (or host function) built into the NEAR runtime to hash the key. To use a custom function,
 /// use [`with_hasher`]. Alternative builtin hash functions can be found at
 /// [`near_sdk::store::key`](crate::store::key).
 ///
+/// # Performance considerations
+/// Note that this collection is optimized for fast removes at the expense of key management.
+/// If the amount of removes is significantly higher than the amount of inserts the iteration
+/// becomes more costly. See [`remove`](FrangibleUnorderedMap::remove) for details.
```
Comment (frol) [inline near-sdk/src/store/frangible_map/mod.rs:34]: As a developer I am left with an unanswered question in my head: "and now what? are there better options?"

File: near-sdk/src/store/unordered_set/mod.rs
Comment (frol) [inline near-sdk/src/store/unordered_set/mod.rs:1]: It looks to me that UnorderedSet has exactly the same problems, isn't it? Should it also be renamed?

**Review by frol — CHANGES_REQUESTED** [review]

File: near-sdk/src/store/unordered_set/mod.rs
Comment (ruseinov) [reply near-sdk/src/store/unordered_set/mod.rs:1]: Yeah, at first I thought that it does not, but now I understand that it does. It also does not have a `defrag` method, but that's easy to fix.

**Comment by ruseinov** [comment]
@frol this will need a test re-run due to a flaky test.

File: near-sdk/src/store/frangible_map/mod.rs (hunk 1 above)
Comment (ruseinov) [reply near-sdk/src/store/frangible_map/mod.rs:34]: Fixed.

**Comment by frol** [comment]
Renaming won't help us to introduce `UnorderedMap` name back (see my https://github.com/near/near-sdk-rs/issues/1134#issuecomment-1936963703). So I don't think renaming is the right move here.

**Comment by ruseinov** [comment]
> Renaming won't help us to introduce `UnorderedMap` name back (see my [#1134 (comment)](https://github.com/near/near-sdk-rs/issues/1134#issuecomment-1936963703)). So I don't think renaming is the right move here.

Why not? Can't we introduce something like `UnorderedMapV2` and type alias it to `UnorederedMap`?

**Comment by ruseinov** [comment]
> Renaming won't help us to introduce `UnorderedMap` name back (see my [#1134 (comment)](https://github.com/near/near-sdk-rs/issues/1134#issuecomment-1936963703)). So I don't think renaming is the right move here.

Right, let's deprecate them then.

File: examples/versioned/src/lib.rs
Comment (frol) [inline examples/versioned/src/lib.rs:1]: I think the changes in this file has to be reverted.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## [Unreleased]
 
+### Fixed
+- deprecated `store::UnorderedMap` due to its fragmentation property ([#1139](https://github.com/near/near-sdk-rs/pull/1139))
+- deprecated `store::UnorderedSet` due to its fragmentation property ([#1139](https://github.com/near/near-sdk-rs/pull/1139))
```
Comment (frol) [inline CHANGELOG.md:7]: We don't need to manually edit CHANGELOG anymore, just write it in the PR description and release-plz will include it in the CHANGELOG, see the automated PR: https://github.com/near/near-sdk-rs/pull/1138

**Review by frol — CHANGES_REQUESTED** [review]

**Comment by ruseinov** [comment]
@frol all fixed.

File: examples/versioned/src/lib.rs
Comment (ruseinov) [reply examples/versioned/src/lib.rs:1]: We should probably make sure this gets caught by the CI though.

**Review by frol — APPROVED** [review]

**Pull request merged**